src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterOperations.cs

```diff
@@ -65,6 +65,15 @@ public async Task AddSearchParameterAsync(ITypedElement searchParam, Cancellatio
                 // verify the parameter is supported before continuing
                 var searchParameterWrapper = new SearchParameterWrapper(searchParam);
                 var searchParameterInfo = new SearchParameterInfo(searchParameterWrapper);
+
+                if (searchParameterInfo.Component?.Any() == true)
+                {
+                    foreach (SearchParameterComponentInfo c in searchParameterInfo.Component)
+                    {
+                        c.ResolvedSearchParameter = _searchParameterDefinitionManager.GetSearchParameter(c.DefinitionUrl.OriginalString);
+                    }
+                }
+
```

Comment:
Does it make sense to add a custom search test for this scenario?

Reply:
Added a unit test to ensure components get resolved